src/attachmentlistmodel.cpp

```diff
@@ -199,44 +195,6 @@ void AttachmentListModel::onMessagesUpdated(const QMailMessageIdList &ids)
     }
 }
 
-static bool findPartFromAttachment(const QMailMessagePart &part, const QString &attachmentLocation, QMailMessagePart &found)
-{
-    if (part.multipartType() == QMailMessagePart::MultipartNone) {
```

Comment:
This multipart thing seems now gone with these changes. Is it still supposed to work as expected? Code here seems like your change a long time ago on commit 252cde6b2

Reply:
I introduced this function to get attachment parts for locations like `1.2.3`. I was a bit silly at that time to write a function myself here, while `QMailMessagePartContainer::partAt()` do exactly this, as far as I know. See `QMailMessagePartContainerPrivate::partAt()` in qmf/src/libraries/qmfclient/qmailmessage.cpp for actual implementation.

So here, instead of delegating the initialisation of `sourcePart` to a custom function, I'm directly instanciating it from a `partAt()` call. Additionally, the code fixed by 252cde6, was already using a `partAt()`call, but was faulty because it was calling the uint API version, which provide access to only one level of multipart nesting, while the `partAt()` with a location API from QMailMessage, can sort out the nesting by itself.

Reply:
Ah right. Had a look earlier but didn't immediately understand that it should be indeed doing nesting.